### DISCUSSION TITLE
Misssing App Key Exception - Portainer

### DISCUSSION THREAD

**Discussion by DSew20** [firefly-iii/firefly-iii#7808] [General]
I'm trying to setup Firefly in docker, using the provided docker-compose including the importer.  I am using Portainer to manage docker and have imported the various .env files.

I had to change the app port in the compose (e.g. to 8033:8080) to avoid a clash.  Within the environments, I have set-up the cron token and matched in the compose, updated the database and db passwords, TZ and added a 32 char APP_KEY.  But, when trying to deploy I get a fatal error as below.  It seems that the APP_KEY is not seen?  Or, is this something else?  I have left the APP_URL as http://localhost and FIRE_FLY_URL as http://app:8080.

Can anybody provide a few pointers as to how to resolve this issue?  Is it the key or perhaps the URLs?

`PHP Fatal error:  Uncaught Illuminate\\Encryption\\MissingAppKeyException: No application encryption key has been specified. in /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php:79\nStack trace:\n#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/helpers.php(307): Illuminate\\Encryption\\EncryptionServiceProvider->Illuminate\\Encryption\\{closure}()\n#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(77): tap()\n#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(60): Illuminate\\Encryption\\EncryptionServiceProvider->key()\n#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(32): Illuminate\\Encryption\\EncryptionServiceProvider->parseKey()\n#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(908): Illuminate\\Encryption\\EncryptionServiceProvider->Illuminate\\Encryption\\{closure}()\n#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(795): Illuminate\\Container\\Container->build()\n#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(933): Illuminate\\Container\\Container->resolve()\n#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): Illuminate\\Foundation\\Application->resolve()\n#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(918): Illuminate\\Container\\Container->make()\n#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1454): Illuminate\\Foundation\\Application->make()\n#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(213): Illuminate\\Container\\Container->offsetGet()\n#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(193): Illuminate\\Session\\SessionManager->buildEncryptedSession()\n#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(78): Illuminate\\Session\\SessionManager->buildSession()\n#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(66): Illuminate\\Session\\SessionManager->createNativeDriver()\n#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Manager.php(106): Illuminate\\Session\\SessionManager->createFileDriver()\n#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Manager.php(80): Illuminate\\Support\\Manager->createDriver()\n#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionServiceProvider.php(52): Illuminate\\Support\\Manager->driver()\n#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(908): Illuminate\\Session\\SessionServiceProvider->Illuminate\\Session\\{closure}()\n#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(795): Illuminate\\Container\\Container->build()\n#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(933): Illuminate\\Container\\Container->resolve()\n#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): Illuminate\\Foundation\\Application->resolve()\n#21 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(918): Illuminate\\Container\\Container->make()\n#22 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1454): Illuminate\\Foundation\\Application->make()\n#23 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/RoutingServiceProvider.php(120): Illuminate\\Container\\Container->offsetGet()\n#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(908): Illuminate\\Routing\\RoutingServiceProvider->Illuminate\\Routing\\{closure}()\n#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(795): Illuminate\\Container\\Container->build()\n#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(933): Illuminate\\Container\\Container->resolve()\n#27 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): Illuminate\\Foundation\\Application->resolve()\n#28 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(918): Illuminate\\Container\\Container->make()\n#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1454): Illuminate\\Foundation\\Application->make()\n#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/RoutingServiceProvider.php(174): Illuminate\\Container\\Container->offsetGet()\n#31 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(908): Illuminate\\Routing\\RoutingServiceProvider->Illuminate\\Routing\\{closure}()\n#32 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(795): Illuminate\\Container\\Container->build()\n#33 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(933): Illuminate\\Container\\Container->resolve()\n#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): Illuminate\\Foundation\\Application->resolve()\n#35 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(918): Illuminate\\Container\\Container->make()\n#36 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(120): Illuminate\\Foundation\\Application->make()\n#37 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(787): app()\n#38 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(658): response()\n#39 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(586): Illuminate\\Foundation\\Exceptions\\Handler->renderHttpException()\n#40 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(492): Illuminate\\Foundation\\Exceptions\\Handler->prepareResponse()\n#41 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(409): Illuminate\\Foundation\\Exceptions\\Handler->renderExceptionResponse()\n#42 /var/www/html/app/Exceptions/Handler.php(173): Illuminate\\Foundation\\Exceptions\\Handler->render()\n#43 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(217): FireflyIII\\Exceptions\\Handler->render()\n#44 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(194): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->renderHttpResponse()\n#45 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(254): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException()\n#46 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()\n#47 {main}\n  thrown in /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php on line 79`

**Comment by JC5** [comment]
No it doesn't seem to pick up on the variables. I don't use portainer myself, are you sure the environment variables reach the container?

**Comment by Kickbut101** [comment]
@DSew20 

You might have to change out the name of the .env files for "stack.env" for portainer to pickup the env variables
![image](https://github.com/firefly-iii/firefly-iii/assets/16029826/975e4c4a-220a-4170-9796-6831a2eb515e)
